### DISCUSSION TITLE
Android15的手机上，将emoji内容设置到PAGText.text后上屏无法显示

### DISCUSSION THREAD

**Discussion by 1456397920** [Tencent/libpag#2461] [Q&A] [answered]
由于保密原因暂不能说具体机型，只能说该手机的系统是Android15，后续条件允许的话帮忙看下

**Comment by kevingpqi123** [comment]
你们看下其它机型是否可以复现，没有复现路径这里没法分析哈

**Reply by 1456397920** [reply]
目前我用官方demo只在Vivo X100手机（Android14）上和该Android15的手机上跑了一遍，X100手机上可以正常显示

**Comment by domchen** [comment]
如果是双字节emoji，就比如各种国旗，要使用release版本里带harfbuzz后缀的版本才可以正常显示，代价是增加包体大小。看看是不是这个问题。

**Reply by 1456397920** [reply]
好的，我用这个版本试试

**Reply by 1456397920** [reply]
使用libpag_4.3.62_android_armeabi_armv7a_arm64v8a_harfbuzz.aar还是有问题，后续等Android15的手机公开，有条件的话帮忙看下，多谢

**Reply by domchen** [reply]
那应该是新系统修改了字体文件路径了。

**Comment by WangShuo1143368701** [comment]
https://android-developers.googleblog.com/2024/07/the-fourth-beta-of-android-15.html
新的系统是有问题，看上面链接 @domchen @kevingpqi123 @imzc @jcyuan 。

**Reply by domchen** [reply]
收到，我们看看

**Reply by 1456397920** [reply]
这个问题目前评估可以解决吗，可以的话大概解决日期是啥时候

**Reply by kevingpqi123** [reply]
我们在 WeTest 上找到了一台 Android 15 的机器，已经复现到这个问题，由于现在还有别的开发任务，我们争取下周抽出一些时间看下

**Reply by 1456397920** [reply]
这个问题能够在11月25号之前解决并release吗，再晚的话我们这里就没有计划版本推送给用户了。。。

**Reply by kevingpqi123** [reply]
我们今天开始看这个问题，同时提供一个解决思路，Android 15 删除了 NotoColorEmojiLegacy.ttf 字体，pag 支持设置回退字体，将这个字体下载设置给 pag 的字体回退列表就可以解决该问题
这样现有的 pag 版本不需要升级就可以正常使用

**Reply by 1456397920** [reply]
这个方案之前评估过，能解决问题，但下载NotoColorEmojiLegacy.ttf字体带来的应用体积增大我们无法接受，所以当时未采纳这个方案

**Reply by kevingpqi123** [reply]
这里可以做成网络下载的，不用放在 APP 中，FreeType 升级会有一些兼容性问题，支持 COLRV1 会有一些改动，看了 skia 这部分的适配，还是有一些工作量的，尽量赶上你们的产品节奏，如果你们比较着急，上面的可以作为保底方案

**Reply by 1456397920** [reply]
了解，那我们先这样尝试解决

**Reply by kevingpqi123** [reply]
目前正在看这部分，升级到最新版本的 freetype 改动量有点大，且 COLRV1是一种新的标准，不仅仅是升级版本就可以解决的，还需要一些时间

如果你们比较着急，先额外注入一个字体，这是可以解决问题的，等我们 freetype 升级后会第一时间同步给大家

**Reply by WangShuo1143368701** [reply]
大概什么时间能推出解决的新版本嘛 @kevingpqi123

**Reply by kevingpqi123** [reply]
最快下周，最慢两周内

**Reply by WangShuo1143368701** [reply]
最新发的版本还没有修复嘛 @kevingpqi123

**Reply by kevingpqi123** [reply]
由于该问题目前有规避方案以及需求优先级的问题，这里的优先级延后了，支持后会第一时间同步给大家

**Reply by WangShuo1143368701** [reply]
麻烦提高一下优先级，下个版本支持下 @kevingpqi123

**Reply by fhbianling** [reply]
> 由于该问题目前有规避方案以及需求优先级的问题，这里的优先级延后了，支持后会第一时间同步给大家

你好，能提一下规避方案是什么吗？我试过直接替换字体为NotoColorEmoji.ttf，可以解决这个问题，但是字体文件太大了，有没有什么更加轻量级的解决方案呢？

**Reply by kevingpqi123** [reply]
将 NotoColorEmoji.ttf 字体通过网络下发，缓存到本地，然后本地注册使用

**Reply by zhuolu09** [reply]
想问下目前修复了吗

**Reply by domchen** [reply]
主干main分支里已经修复了，现在Android平台支持系统emoji的读取了。但是这个版本要发出来可能没那么快。

**Reply by zhuolu09** [reply]
有预估的发布时间吗😶‍🌫️

**Reply by domchen** [reply]
暂时还没有，下个版本架构变化特别大。不支持OpenGL ES 2.0了。另外还有新的基于XML文本的PAGX格式设计，方便AI输入输出。短期确实没那么快。我们看看有没可能在之前版本的tgfx上把这个特性单独移植过去。尽快发个包

**Reply by zhuolu09** [reply]
好的，感谢🙏

**Reply by kevingpqi123** [reply, marked as answer]
可以升级到 4.5.27，该版本开始已支持 Android 原生  emoji 字体

**Reply by zhuolu09** [reply]
好的👌
